src/puppetlabs/trapperkeeper/services/metrics/metrics_core.clj

```diff
@@ -34,24 +34,27 @@
 (def MetricsServiceContext
   {:registries (schema/atom {schema/Any RegistryContext})})
 
+(def Keyword-or-Str (schema/if keyword? schema/Keyword schema/Str))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Private
 
 (schema/defn jmx-reporter :- JmxReporter
   [registry :- MetricRegistry
-   domain :- (schema/maybe schema/Str)]
+   domain :- (schema/maybe schema/Keyword)]
   (let [b (JmxReporter/forRegistry registry)]
-    (when-let [^String d domain]
-      (.inDomain b d))
+    (when-let [d domain]
+      (.inDomain b ^String (name d)))
```

Comment:
I would figure that clojure wouldn't need the type hint here since `name` always outputs a string, right? Genuine question, I'm not sure